cscs-checks/microbenchmarks/hpcg/hpcg_benchmark.py

```diff
@@ -0,0 +1,148 @@
+
+    def setup(self, partition, environ, **job_opts):
+        self.num_tasks = self.system_num_tasks[partition.fullname]
+
+        super().setup(partition, environ, **job_opts)
+
+
+@rfm.required_version('>=2.16-dev0')
+@rfm.simple_test
+class HPCGCheckMKL(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.descr = 'HPCG benchmark Intel MKL implementation'
+        self.valid_systems = ['daint:mc', 'dom:mc', 'daint:gpu', 'dom:gpu']
+        self.valid_prog_environs = ['PrgEnv-intel']
+        self.modules = ['craype-hugepages8M']
+        #self.sourcesdir needed for "CrayXC" config file
+        self.build_system = 'Make'
+        self.prebuild_cmd = ['cp -r ${MKLROOT}/benchmarks/hpcg/* .',
+                             'mv Make.CrayXC setup',
+                             './configure CrayXC']
+
+        self.num_tasks = 0
+        self.num_tasks_per_core = 2
+        self.problem_size = 104
+
+        self.variables = {
+            'HUGETLB_VERBOSE': '0',
+            'MPICH_MAX_THREAD_SAFETY' : 'multiple',
+            'MPICH_USE_DMAPP_COLL': '1',
+            'PMI_NO_FORK': '1',
+            'KMP_HW_SUBSET': '9c,2t',
+            'KMP_AFFINITY': 'granularity=fine,compact'
+        }
+
+        self.executable = 'bin/xhpcg_avx2'
+        self.executable_opts = ['--nx=%d' % self.problem_size,
+                                '--ny=%d' % self.problem_size,
+                                '--nz=%d' % self.problem_size, '-t2']
+        self.sanity_patterns = sn.assert_eq(4, sn.count(
+            sn.findall(r'PASSED', self.outfile_lazy)))
+        self.reference = {
+            'dom:mc': {
+                'gflops': (22, -0.1, None, 'GFLOPs')
+            },
+            'daint:mc': {
+                'gflops': (22, -0.1, None, 'GFLOPs')
+            },
+            'dom:gpu': {
+                'gflops': (10.7, -0.1, None, 'GFLOPs')
+            },
+            'daint:gpu': {
+                'gflops': (10.7, -0.1, None, 'GFLOPs')
+            },
+        }
+
+        self.maintainers = ['SK']
+        self.tags = {'diagnostic'}
+
+    @property
+    @sn.sanity_function
+    def num_tasks_assigned(self):
+        return self.job.num_tasks
+
+    @property
+    @sn.sanity_function
+    def outfile_lazy(self):
+        pattern = 'n%d-%dp-%dt-*.yaml' % (self.problem_size,
+                                          self.job.num_tasks,
+                                          self.num_cpus_per_task)
+        return sn.getitem(sn.glob(pattern), 0)
+
+    def setup(self, partition, environ, **job_opts):
+        if partition.fullname in ['daint:gpu', 'dom:gpu']:
+            self.num_tasks_per_node = 2
+            self.num_cpus_per_task = 12
+        else:
+            self.num_tasks_per_node = 4
+            self.num_cpus_per_task = 18
+
+        # since this is a flexible test, we divide the extracted
+        # performance by the number of nodes and compare
+        # against a single reference
```

Comment:
The number of tasks_per_node is different between the tests.
While `--flex-alloc-tasks=8` would run on 4 nodes on daint:gpu and 2 nodes daint:mc for the MKL case, the reference implementation has to have at least 36 tasks.
If there was a `--flex-alloc-nodes`, it would work. Or I split the test in two files.
In fact, there should be a check in the test that throws an error if the allocated tasks
is not a multiple of `num_tasks_per_node`

Reply:
This I understand. My question is basically, why the base test is not flexible as well?